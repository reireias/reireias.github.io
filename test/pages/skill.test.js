import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vue from 'vue'
import Vuetify from 'vuetify'
import CompositionApi from '@vue/composition-api'
import target from '@/pages/skill'

Vue.use(Vuetify)
const localVue = createLocalVue()
localVue.use(Vuetify)
localVue.use(CompositionApi)

describe('skills', () => {
  it('should be vue instance', () => {
    const wrapper = shallowMount(target, {
      localVue,
      Vuetify,
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
