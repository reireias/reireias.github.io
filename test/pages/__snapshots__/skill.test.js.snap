// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`skills should be vue instance 1`] = `
"<div class=\\"skill-page\\">
  <page-title-stub title=\\"Skill\\"></page-title-stub>
  <v-expansion-panels-stub activeclass=\\"v-item--active\\" multiple=\\"true\\" class=\\"panel\\">
    <v-expansion-panel-stub>
      <v-expansion-panel-header-stub expandicon=\\"$expand\\" class=\\"title\\">システムアーキテクト</v-expansion-panel-header-stub>
      <v-expansion-panel-content-stub>
        <p>可用性、コスト、拡張性、運用などを考慮したシステム全体のアーキテクチャ設計を行うことが得意です。</p>
        <p>プラットフォームはAWS、GCP、自社データセンター等を利用してきました。OSはLinux系を一通り構築に使ったことがあります。</p>
        <p>最近はAWSのマネージドサービスを活用し極力運用負荷が少なく、かつ、過度に複雑にならないシステム設計とする場合が多いです。</p>
        <p>過去に業務で構築した経験のある構成は次の通りです。</p>
        <p>ELB + EC2 + RDS</p>
        <p>ECS(Fargate)を利用したコンテナベースのアプリケーション</p>
        <p>API Gateway + LambdaやCloudFront + Lambda Edge等のサーバーレス構成</p>
        <p>SQS + SNS + Lambdaを利用したキューチェーン型システム</p>
      </v-expansion-panel-content-stub>
    </v-expansion-panel-stub>
    <v-expansion-panel-stub>
      <v-expansion-panel-header-stub expandicon=\\"$expand\\" class=\\"title\\">Infrastracture as Codeの実現</v-expansion-panel-header-stub>
      <v-expansion-panel-content-stub>
        <p>Infrastracture as Codeによる利点と恩恵を考慮した上で、管理/変更/レビュー/運用のしやすいInfrastracture as Codeな世界を実現することができます。</p>
        <p>ツールとしてはTerraformを利用することが多いですが、CloudFormationによる実装経験もあります。</p>
        <p>Terraformに関してはTerraform Cloudによるplan/applyの自動化や、AWS/GCP以外にGitHubやTerraform Cloud自体の管理などヘビーに活用しています。</p>
      </v-expansion-panel-content-stub>
    </v-expansion-panel-stub>
    <v-expansion-panel-stub>
      <v-expansion-panel-header-stub expandicon=\\"$expand\\" class=\\"title\\">CI/CDパイプラインの構築</v-expansion-panel-header-stub>
      <v-expansion-panel-content-stub>
        <p>CIに関しては、GitHub Actions/CircleCI/Travis CI等の使用経験があり、Linux + Dockerの知識も合わせて適切な設定を行うことができます。</p>
        <p>CDに関しては主にAWSのCodeシリーズを利用しての構築が得意です。</p>
        <p>以下に業務で実際に行ってきたものを列挙します。</p>
        <p>- まれに落ちるテストや時間のかかるテストによるCI詰まりなどの分析/修正</p>
        <p>- Blue/Greenデプロイメントやロールバック手順の確立</p>
        <p>- AWS SSMパラメータストアやKMS等を利用した秘匿情報の適切な管理</p>
        <p>- Dockerイメージビルドの高速化</p>
        <p>- CI/CDパイプラインにおける適切なキャッシュ設定</p>
      </v-expansion-panel-content-stub>
    </v-expansion-panel-stub>
    <v-expansion-panel-stub>
      <v-expansion-panel-header-stub expandicon=\\"$expand\\" class=\\"title\\">運用系ツールの作成と運用</v-expansion-panel-header-stub>
      <v-expansion-panel-content-stub>
        <p>もともとサービスのフロントエンド/バックエンドの開発が長く、現在でも社内のリソース状況に応じてアプリケーションの実装を担当したり、趣味での実装の多くがWebアプリケーション開発なので、実はインフラと同等以上にアプリケーション開発のスキルもあったりします。</p>
        <p>これを活かし、省リソース(要はワンオペ)での管理機能等の小さいサービスのインフラとアプリケーション両方の開発や、日々の運用で利用するような夜間停止やSlack連携などの軽いLambdaアプリケーション等、開発時の便利系スクリプト等の実装を担当することが多くありました。</p>
      </v-expansion-panel-content-stub>
    </v-expansion-panel-stub>
    <v-expansion-panel-stub>
      <v-expansion-panel-header-stub expandicon=\\"$expand\\" class=\\"title\\">Data WarehouseとBI Toolでの分析/可視化基盤の作成</v-expansion-panel-header-stub>
      <v-expansion-panel-content-stub>
        <p>ビッグデータ解析とまではいかないですが、それなりに多量のデータを格納し、分析/可視化するための基盤の構築も何度か経験があります。</p>
        <p>データ量やリアルタイム性、分析する対象に応じて適切なデータ格納方法やデータパイプラインを構築し、ビジネスメンバーの興味関心に応じたダッシュボード作成等を実践してきました。</p>
        <p>個人的にはエンジニアサイドでSQL等のクエリを実行できるBI環境を提供し、ビジネスサイドでKPIや施策に応じたクエリを日々作成していくスタイルが理想です。</p>
      </v-expansion-panel-content-stub>
    </v-expansion-panel-stub>
    <v-expansion-panel-stub>
      <v-expansion-panel-header-stub expandicon=\\"$expand\\" class=\\"title\\">PaaS/Docker/低レイヤー技術の社内教育</v-expansion-panel-header-stub>
      <v-expansion-panel-content-stub>
        <p>数年前からは社内の若手エンジニアへAWSやDocker、Linux、ネットワーク等の社内勉強会を主催したりもしています。</p>
        <p>主に入門〜中級くらいの内容で、社内全体のスキルの底上げや、サービス開発/運用時におけるデバッグやバグ修正の補助となるような内容で実施しています。</p>
        <p>また、理解度の面からハンズオン形式にこだわって実施しています。</p>
      </v-expansion-panel-content-stub>
    </v-expansion-panel-stub>
    <v-expansion-panel-stub>
      <v-expansion-panel-header-stub expandicon=\\"$expand\\" class=\\"title\\">Webアプリケーション開発のリーディング</v-expansion-panel-header-stub>
      <v-expansion-panel-content-stub>
        <p>過去にBtoBやBtoCを含め、6プロダクトほどサービス開発を経験してきました。</p>
        <p>セキュリティや可用性、性能等の非機能要件を考慮した開発や、メンバーのスキルや学習コスト/メンテナンスコストを考慮した技術選択等を実践できます。</p>
        <p>認証や非同期処理、大規模システムの結合ビルドの調整等、比較的難易度の高い部分を担当することが多いです。</p>
        <p>最近は若手層への教育を行いながら開発リーダーを務めることが多いです。(スクラムマスター、プロダクトオーナーではないので、開発外部との交渉/調整等は専門外ですが)</p>
        <p>また、IPAの資格で応用情報技術者、セキュリティスペシャリスト、データベーススペシャリストを保有しています。</p>
        <p>1年以上業務で利用経験のある言語: Ruby, Java, C#</p>
        <p>同フレームワーク: Ruby on Rails, Spring, Vue.js</p>
      </v-expansion-panel-content-stub>
    </v-expansion-panel-stub>
    <v-expansion-panel-stub>
      <v-expansion-panel-header-stub expandicon=\\"$expand\\" class=\\"title\\">開発環境の構築や改善</v-expansion-panel-header-stub>
      <v-expansion-panel-content-stub>
        <p>Webアプリケーションと本番環境のインフラ構成に加え、DockerやLinux等の知識があるため、開発環境まわりの構築や改善を実施してきました。</p>
        <p>これらの知識を組み合わせることで、本番環境をなるべく再現しつつも起動が高速で扱いやすいローカルの開発環境を構築したり、クレデンシャル/API</p>
        <p>Tokenのセキュリティや職能/職種に合わせた権限設定を考慮したリポジトリや成果物管理の仕組み等、開発環境とその周辺を構築してきました。</p>
        <p>技術的にこれらを構築するだけでなく、開発フローやブランチ戦略の考案/改善といったことも得意だったりします。</p>
      </v-expansion-panel-content-stub>
    </v-expansion-panel-stub>
    <v-expansion-panel-stub>
      <v-expansion-panel-header-stub expandicon=\\"$expand\\" class=\\"title\\">資格</v-expansion-panel-header-stub>
      <v-expansion-panel-content-stub>
        <p>AWS ソリューションアーキテクトアソシエイト</p>
        <p>IPA 応用情報技術者</p>
        <p>IPA データベーススペシャリスト</p>
        <p>IPA セキュリティスペシャリスト</p>
      </v-expansion-panel-content-stub>
    </v-expansion-panel-stub>
  </v-expansion-panels-stub>
</div>"
`;
